backend/src/ee/controllers/v1/machineIdentitiesController.ts

```diff
@@ -0,0 +1,802 @@
+    }
+
+    if (!validatedClientSecretDatum) throw UnauthorizedRequestError();
+
+    const {
+        clientSecretTTL,
+        clientSecretNumUses,
+        clientSecretNumUsesLimit,
+    } = validatedClientSecretDatum;
+
+    if (clientSecretTTL > 0) {
+        const clientSecretCreated = new Date(validatedClientSecretDatum.createdAt)
+        const ttlInMilliseconds = clientSecretTTL * 1000;
+        const currentDate = new Date();
+        const expirationTime = new Date(clientSecretCreated.getTime() + ttlInMilliseconds);
+
+        if (currentDate > expirationTime) {
+            await MachineIdentityClientSecret.findByIdAndUpdate(
+                validatedClientSecretDatum._id,
+                {
+                    isActive: false
+                }
+            );
+
+            throw UnauthorizedRequestError({
+                message: "Failed to authenticate MI credentials due to expired client secret"
+            });
+        }
+    }
+
+    if (clientSecretNumUses > 0 && clientSecretNumUses === clientSecretNumUsesLimit) {
+        // number of times client secret can be used for 
+        // a login operation reached
+        await MachineIdentityClientSecret.findByIdAndUpdate(
+            validatedClientSecretDatum._id,
+            {
+                isActive: false
+            },
+            {
+                new: true
+            }
+        );
+
+        throw UnauthorizedRequestError({
+            message: "Failed to authenticate MI credentials due to client secret number of uses limit reached"
+        });
+    }
+
+    // increment usage count by 1
+    await MachineIdentityClientSecret.findByIdAndUpdate(
+        validatedClientSecretDatum._id,
+        {
+            $inc: { clientSecretNumUses: 1 }
+        },
+        {
+            new: true
+        }
+    );
+
+    // token version
+    const accessToken = createToken({
+        payload: {
+            machineId: machineIdentity._id.toString(),
+            clientSecretDataId: validatedClientSecretDatum._id.toString(),
+            authTokenType: AuthTokenType.MACHINE_ACCESS_TOKEN,
+            tokenVersion: validatedClientSecretDatum.accessTokenVersion
+        },
+        expiresIn: machineIdentity.accessTokenTTL,
+        secret: await getAuthSecret()
+    });
+
+    const userAgent = req.headers["user-agent"] ?? "";
+
+    await EEAuditLogService.createAuditLog(
+        {
+            actor: {
+                type: ActorType.MACHINE,
+                metadata: {
+                    machineId: machineIdentity._id.toString(),
+                    name: machineIdentity.name
+                }
+            },
+            authPayload: machineIdentity,
+            ipAddress: req.realIP,
+            userAgent,
+            userAgentType: getUserAgentType(userAgent)
+        },
+        {
+            type: EventType.LOGIN_MACHINE_IDENTITY,
+            metadata: {
+                machineId: machineIdentity._id.toString(),
+                clientId,
```

Comment:
I would remove clientId because it's half the puzzle to machine login. I think clientSecretId is enough